src/servers/multi_server.cc

```diff
@@ -0,0 +1,973 @@
+  TRITONSERVER_ServerOptions *server1_options = nullptr,
+                             *server2_options = nullptr;
+  SetServerOptions(
+      &server1_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[1]);
+  SetServerOptions(
+      &server2_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[2]);
+
+  TRITONSERVER_Server *server1_ptr = nullptr, *server2_ptr = nullptr;
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server1_ptr, server1_options),
+      "creating first server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server2_ptr, server2_options),
+      "creating second server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server1_options),
+      "deleting server options");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server2_options),
+      "deleting server options");
+
+  std::shared_ptr<TRITONSERVER_Server> server1(
+      server1_ptr, TRITONSERVER_ServerDelete);
+  std::shared_ptr<TRITONSERVER_Server> server2(
+      server2_ptr, TRITONSERVER_ServerDelete);
+
+  // Wait and until the servers are both live and ready.
+  CheckServerLiveAndReady(server1);
+  CheckServerLiveAndReady(server2);
+
+  // Print status of the servers.
+  PrintServerStatus(server1);
+  PrintServerStatus(server2);
+
+  // Load models in both servers.
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server1.get(), "simple1"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server1.get(), "simple2"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server2.get(), "simple1"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server2.get(), "simple3"),
+      "failed to load model");
+
+  // Wait for the models to become available.
+  bool is_int1 = true, is_int2 = true, is_int3 = true;
+  bool is_torch_model1 = false, is_torch_model2 = false,
+       is_torch_model3 = false;
+  AwaitModelReady(server1, "simple1", &is_int1, &is_torch_model1);
+  AwaitModelReady(server1, "simple2", &is_int2, &is_torch_model2);
+  AwaitModelReady(server2, "simple1", &is_int1, &is_torch_model1);
+  AwaitModelReady(server2, "simple3", &is_int3, &is_torch_model3);
+
+  // Create the allocator that will be used to allocate buffers for
+  // the result tensors.
+  TRITONSERVER_ResponseAllocator* allocator = nullptr;
+  FAIL_IF_ERR(
+      TRITONSERVER_ResponseAllocatorNew(
+          &allocator, ResponseAlloc, ResponseRelease, nullptr /* start_fn */),
+      "creating response allocator");
+
+  // Inference
+  RunInferenceAndValidate(
+      server1, allocator, "simple1", is_int1, is_torch_model1);
+  RunInferenceAndValidate(
+      server1, allocator, "simple2", is_int2, is_torch_model2);
+  RunInferenceAndValidate(
+      server2, allocator, "simple1", is_int1, is_torch_model1);
+  RunInferenceAndValidate(
+      server2, allocator, "simple3", is_int3, is_torch_model3);
```

Comment:
Also add test on operating on the wrong model? i.e. load/infer/unload simple3 in server1, to make sure different servers are monitoring different repositories.

Reply:
Good catch. I will add this